p2p/protocols/protocol.go

```diff
@@ -288,40 +384,46 @@ func (p *Peer) Send(ctx context.Context, msg interface{}) error {
 	return p2p.Send(p.rw, code, wmsg)
 }
 
-// handleIncoming(code)
-// is called each cycle of the main forever loop that dispatches incoming messages
-// if this returns an error the loop returns and the peer is disconnected with the error
-// this generic handler
-// * checks message size,
-// * checks for out-of-range message codes,
-// * handles decoding with reflection,
-// * call handlers as callbacks
-func (p *Peer) handleIncoming(handle func(ctx context.Context, msg interface{}) error) error {
+// Receive(code) is a sync call that handles incoming message with provided message handler
+func (p *Peer) Receive(handler func(ctx context.Context, msg interface{}) error) error {
```

Comment:
sure we want this exported?

Reply:
Nope, good catch. We can export it if needed, later